packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -81,6 +87,18 @@ const listLocationItemsAction = createEnhancedListHandler(
   listLocationItemsHandler
 );
 
+const getDownloadErrorMessageFromFailedDownloadTask = (
+  tasks: Tasks<DownloadHandlerData>
+): string | undefined => {
+  if (!tasks.length) {
+    return undefined;
+  }
+
+  return `Failed to download ${
+    tasks[0].data.fileKey ?? tasks[0].data.key
+  } due to error: ${tasks[0].message}.`;
+};
```

Comment:
We will need to readdress as this approach doesn't align with the display text api, nor should hasDownloadError be exposed in the hook API but am curious, does download not provide an error message?